### PR TITLE
New module: azure_rm_ipgroup

### PR DESCRIPTION
**Pull request by ghost**
##### SUMMARY
This module supports following operations for IP Group resource -

1. Create new IP Group
2. Update an existing IP group
3. Delete IP group
4. Get details about specific IP Group
5. List all IP groups by resource group or subscription

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_ipgroup
azure_rm_ipgroup_info

ADDITIONAL INFORMATION
https://docs.microsoft.com/en-us/rest/api/virtualnetwork/ipgroups

### CODE REVIEW

File: plugins/modules/azure_rm_ipgroup_info.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+        # store the results of the module operation
+        self.results = dict(
+            changed=False
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+        self.log_path = None
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ipgroup_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_ipgroup_info.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        self.log_mode = None
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ipgroup_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_ipgroup.py [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.location = None
+        self.ip_addresses = None
+        self.tags = None
+        self.log_path = None
+        self.log_mode = None
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ipgroup.py]:
```suggestion

```

File: plugins/modules/azure_rm_ipgroup.py [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+            state=dict()
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.location = None
+        self.ip_addresses = None
+        self.tags = None
+        self.log_path = None
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_ipgroup.py]:
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
@techcon65 It will better to add check mode test for "azure/tests/integration/targets/azure_rm_ipgroup/tasks/main.yml"
```e.g.
- name: Create IP group (check mode)
  azure_rm_ipgroup:
    resource_group: "{{ resource_group }}"
    name: "{{ group_name }}"
    location: eastus
    ip_addresses:
      - 13.64.39.16/32
      - 40.74.146.80/31
      - 40.74.147.32/28
    tags:
      key1: "value1"
    state: present
  check_mode: yes

**Comment by ghost** [comment]
> @techcon65 It will better to add check mode test for "azure/tests/integration/targets/azure_rm_ipgroup/tasks/main.yml"
> 
> ```
> - name: Create IP group (check mode)
>   azure_rm_ipgroup:
>     resource_group: "{{ resource_group }}"
>     name: "{{ group_name }}"
>     location: eastus
>     ip_addresses:
>       - 13.64.39.16/32
>       - 40.74.146.80/31
>       - 40.74.147.32/28
>     tags:
>       key1: "value1"
>     state: present
>   check_mode: yes
> ```

Sure @Fred-sun Added check_mode test as suggested and tested it locally! Working as expected. Thank you very much!

**Pull request merged**